src/main/java/com/ericsson/ei/rules/RulesHandler.java

```diff
@@ -149,113 +148,48 @@ public RulesObject getRulesForEvent(String event) {
-        } catch (Exception e) {
-            LOGGER.error(e.getMessage(), e);
-        }
-        return null;
-    }
-
-    /**
-     * Sets the path where the rules file is located.
-     *
-     * @param path
-     *            the rules file path
-     * @throws IOException
-     */
-    private void setRulesFilePath(String path) throws IOException {
-        if (isPathURL(path)) {
-            rulesFilePath = getRulesFilePath();
-        } else {
-            rulesFilePath = path;
-        }
-    }
-
-    /**
-     * If the path is a URL it will be downloaded and placed into the home folder.
-     *
-     * @param path
-     *            the rules file path
-     * @throws IOException
-     */
-    private void downloadRulesFile(String path) throws IOException {
-        if (isPathURL(path)) {
-            downloadRuleFileToHomeDirectory(path);
-        }
-    }
-
-    /**
-     * Downloads rules file from URL to rules file path.
+     * Reads the rules file content from a URI.
      *
-     * @param url
-     *            the url of rules file
+     * @return rules file content
      * @throws IOException
+     * @throws URISyntaxException
      */
-    private void downloadRuleFileToHomeDirectory(String url) throws IOException {
-        URL source = new URL(url);
-        final File destination = new File(rulesFilePath);
-        FileUtils.copyURLToFile(source, destination);
+    private String readRulesFileFromURI() throws IOException, URISyntaxException {
+        return IOUtils.toString(new URI(rulesFilePath), "UTF-8");
     }
 
     /**
-     * Reads the content of the rules file.
+     * Reads the rules file content from a full or relative path.
      *
      * @return rules file content
+     * @throws IOException
      * @throws Exception
      */
-    private String readRulesJsonFileContent() throws Exception {
+    private String readRulesFileFromPath() throws IOException {
         String rulesJsonFileContent = null;
         InputStream inputStream = this.getClass().getResourceAsStream(rulesFilePath);
         if (inputStream == null) {
             rulesJsonFileContent = FileUtils.readFileToString(new File(rulesFilePath), Charset.defaultCharset());
         } else {
-            rulesJsonFileContent = getContent(inputStream);
-        }
-        if (rulesJsonFileContent.equals("")) {
-            throw new Exception("Rules content cannot be empty");
+            rulesJsonFileContent = IOUtils.toString(inputStream, "UTF-8");
         }
         return rulesJsonFileContent;
     }
 
     /**
-     * Checks if the given path is a URL.
+     * Checks if the path is using a scheme.
      *
      * @param path
      *            the rules file path
-     * @return
+     * @return true if there is a match, false otherwise
      */
-    private boolean isPathURL(String path) {
-        String protocolRegex = "https?";
+    private boolean isPathUsingScheme(String path) {
+        String schemeRegex = "https?|file";
```

Comment:
What it is "file"? Is it the Prefix you are checking, "file://...."? In that case a "/" is valid path as well.

Reply:
The regex only matches against the scheme/protocal part of the string, so no slashes I suppose.

Reply:
So this if statement will fail if user writes the path without "path:/..."  ?,,, Its completely valid to provide unix paths with starting only slash '/' , e.g. : /tmp/myExtractuinRules.json.

Reply:
No, if no scheme is provided it assumes a path has been given.

Reply:
Ok,,
But the java lib that open file path,, doesn't it handle both "file://path/rules.json" and "/path/rules.json" ?
If its not, then why not append prefix "file:/" to file paths with "/" and use same URI open lib for all situations?

Reply:
Well paths can be relative and used to open resource files as well. You can check the logic for that in the same class. I tries to open paths in two different ways.

Reply:
ok